operator/controllers/lifecycle/keptntask/job_utils.go

```diff
@@ -146,18 +133,28 @@ func (r *KeptnTaskReconciler) generateJob(ctx context.Context, task *klcv1alpha3
 		ConfigMap:     definition.Status.Function.ConfigMap,
 	}
 
-	builder := getJobRunnerBuilder(builderOpt)
+	builder := NewJobRunnerBuilder(builderOpt)
 	if builder == nil {
 		return nil, controllererrors.ErrNoTaskDefinitionSpec
 	}
 
-	container, volumes, err := builder.CreateContainerWithVolumes(ctx)
+	container, err := builder.CreateContainer(ctx)
+	if err != nil {
+		r.Log.Error(err, "could not create container for Job")
+		return nil, controllererrors.ErrCannotMarshalParams
+	}
+
+	volume, err := builder.CreateVolume(ctx)
 	if err != nil {
-		r.Log.Error(err, "could not create Job")
+		r.Log.Error(err, "could not create volume for Job")
```

Comment:
I would suggest to rather return a wrapped error, like `fmt.Errorf("could not create volume for Job: %w", err)` and remove the `r.Log` call from here, as any errors will be logged in the `Reconcile` method of the controller.